### PR TITLE
Pick the URL for nameframe only once per slot.

### PR DESCRIPTION
**Pull request by tdrl**

### CODE REVIEW

File: extensions/amp-a4a/0.1/amp-a4a.js [hunk 1]
```diff
@@ -308,6 +308,14 @@ export class AmpA4A extends AMP.BaseElement {
...
      */
     this.isCollapsed_ = false;
+
+    // Note: This URL needs to be per-slot because we don't want ads in
+    // different slots to be able to interfere with each other, so they
+    // need to be cross-domain from each other.  However, we want the
+    // preconnect URL to be the same as the URL used in the fetch step
+    // (obviously), so generate it once here.
+    /** @const {string} */
+    this.nameframeUrl_ = getDefaultBootstrapBaseUrl(this.win, 'nameframe');
```
Comment (keithwrightbos) [inline extensions/amp-a4a/0.1/amp-a4a.js]: Note that we do not reset the nameframeUrl_ as part of unlayoutCallback.  I think this is fine but wanted to point it out.

Comment (tdrl) [inline extensions/amp-a4a/0.1/amp-a4a.js]: Yeah, you're right.  It _shouldn't_ be an issue, but doesn't hurt to be paranoid.

**Comment by tdrl** [comment]
This got more complicated than expected -- we should be not just preconnecting, but prefetching, and I need to write tests etc.  This will take a bit longer than expected.

**Comment by lannka** [comment]
Please remove the WIP label once ready for review. I saw there're tests being commented out.

**Comment by lannka** [comment]
feel free to reopen if you're still on it.

**Pull request closed without merging**